Updates notices / banners for 6.8 release

roccotripaldi opened the pull request:
See screenshots for
- Post-activation banner: #10638
- Post-upgrade banner: #10658


#### Changes proposed in this Pull Request:

* Adds new messaging / images for the post-activation banner and the post-upgrade banner

#### Testing instructions:

1. Spin up a new Jurassic Ninja site with this branch
2. Activate Jetpack and observe the messaging
3. Connect jetpack and observe the messaging

roccotripaldi committed:
Post Activation message:

updated with new content from #10638

roccotripaldi committed:
Upgrade Notice Content

Adding new images and graphics from #10658

roccotripaldi committed:
Adds 2 new images for post upgrade messaging for the 6.8 release.

roccotripaldi committed:
bump version on the upgrade notice

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -47,212 +40,29 @@ const UpgradeNoticeContent = moduleSettingsForm(
 					<p>
-						{ __( 'This release of Jetpack brings with it a major new free feature!' ) }
+						{ __( 'The features you rely on, adapted for the new WordPress editor.' ) }<br />
+						{ __( 'A new editor? Yes! Learn more.' ) }
```
Should there be a link on the "Learn more" text here?

mapk replied:
I've opted to have the sentences "A new editor? Yes! Learn more." removed completely. I'm not too sure where that link would link to. Let's just remove it.

MichaelArestad replied:
> I'm not too sure where that link would link to. Let's just remove it.

+1

mapk replied:
Actually let's just link this to the Jetpack.com announcement post. I've got the permanent url when ready.

mapk replied:
I added the link in https://github.com/Automattic/jetpack/pull/10672/commits/6b5795fe4c16da4312204ee5e341c6659e4f84dd

jeherve replied:
I made some additional changes in d0a4374348fd1d4932194161fd714e38ac8434fe to fix the link syntax and the link URL.

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -47,212 +40,29 @@ const UpgradeNoticeContent = moduleSettingsForm(
-						{ __( 'Turn on one or both and see decreased page load speeds, ' +
-						'as well as reduced bandwidth usage—which may lead to lower hosting costs.'
+						{ __( 'Today, we are introducing the first wave of Jetpack-specific blocks built specifically ' +
+							'for the new editor experience: payment buttons, contact forms, maps, and markdown.'
```
Should some of those things be capitalized?

mapk replied:
I think you're right! Let's capitalize "Simple Payments button, Contact form, Map, and Markdown."

mapk replied:
Added capitalizations in https://github.com/Automattic/jetpack/pull/10672/commits/6b5795fe4c16da4312204ee5e341c6659e4f84dd

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -47,212 +40,29 @@ const UpgradeNoticeContent = moduleSettingsForm(
-					</div>
-
 					<div className="jp-dialogue__cta-container">
 						<Button
```
I think you will need to update this button as well; it currently links to the site accelerator doc.

jeherve replied:
We could maybe link to our release post here?
https://wp.me/p1moTy-cee

jeherve replied:
Since the link text was updated to "Take me to the new editor", I updated the link to point to the post editor in d0cbb3f09cb36dc5eeb9fc441e0ccb7028e59fec

jeherve commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -261,15 +71,19 @@ const UpgradeNoticeContent = moduleSettingsForm(
 						>
 							{ __( 'Learn more' ) }
 						</Button>
+						<Button onClick={ this.dismissNotice }>
```
The 2 buttons seem a bit too close to each other right now:

<img width="897" alt="screenshot 2018-11-20 at 12 12 23" src="https://user-images.githubusercontent.com/426388/48770269-01075800-ecbe-11e8-9df6-c861c1402e32.png">

mapk replied:
Agreed. Let's add 8px between them. And the "Learn more" should be changed to "Take me to the new editor" which should open a new post.

mapk replied:
Updated the button's text in https://github.com/Automattic/jetpack/pull/10672/commits/6b5795fe4c16da4312204ee5e341c6659e4f84dd

jeherve replied:
Added some margin in a02ab4d8a613f3d427e53fa6003046ddc7067566

jeherve reviewed (changes requested):
While the post-activation screen works for me, there are still a few issues with the post-upgrade modal imo.

mapk committed:
Fixed capitalization on block names, added link to 'learn more', and fixed wording on primary button.

roccotripaldi commented:
thanks for adding in all the changes. This probably needs a rebase.

jeherve committed:
Upgrade notice: add missing import and fix link.

- Fix link syntax.
- Use shortlink instead of full permalink as that permalink can change in the future.

jeherve committed:
Upgrade notice: add margin around the buttons.

@see https://github.com/Automattic/jetpack/pull/10672#discussion_r235037576

jeherve committed:
Upgrade notice: learn more link leads to post editor.

jeherve reviewed (dismissed):
This works well for me now.

<img width="817" alt="screenshot 2018-11-20 at 19 01 33" src="https://user-images.githubusercontent.com/426388/48797344-32a21280-ed02-11e8-875d-a2cc0ea73a22.png">

@roccotripaldi Would you mind taking a look at my changes and merge if you're happy with this PR?

jeherve committed:
Update notice: change the URL of the editor if 5.0 isn't installed

If WP version is less than 5.0, we link to wp-admin/update-core.php

jeherve committed:
Upgrade notice: display a different button when WP < 5.0

If someone does not use WP 5.0 yet, we invite them to update instead.
I updated the Tracks actions to reflect that as well.

jeherve commented:
@roccotripaldi 	@dereksmart Could I get your reviews on this? I updated the modal to show a different button depending on the version of WP you currently use.

See p1542742001072600-slack-jetpack-gutenberg for more information.

dereksmart commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -6,20 +6,14 @@ import React from 'react';
 import { translate as __ } from 'i18n-calypso';
 import Button from 'components/button';
-import { connect } from 'react-redux';
+import ExternalLink from 'components/external-link';
```
Thank you for this

mapk committed:
Fixed naming of blocks as @MichaelArestad suggested.

mapk commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -47,219 +48,70 @@ const UpgradeNoticeContent = moduleSettingsForm(
-							'offered speedier and optimized images served from our global Content Delivery Network. ' +
-							'Now we can also speed up your site by serving your static files (think CSS and JavaScript) from the same network.'
+						{ __( 'Today, we are introducing the first wave of Jetpack-specific blocks built specifically ' +
+							'for the new editor experience: Simple Payment button, Contact form, Maps, and Markdown.'
```
Fixed with https://github.com/Automattic/jetpack/pull/10672/commits/fa55df65991826bedb1e5fe85a6e6dc959cda2a4

dereksmart commented on _inc/client/components/upgrade-notice-content/index.jsx:
```diff
@@ -47,219 +48,70 @@ const UpgradeNoticeContent = moduleSettingsForm(
-					from: 'warm_welcome'
-				} );
+		renderLearnMore = () => {
+			const versionSupportsGutenberg = /(5\.0).*/;
```
Rather than checking for version, what if we instead passed `Jetpack_Gutenberg::is_gutenberg_available()` over with the initial state? The reason I suggest this is to catch the cases where someone might be running 4.9 with the Gutenberg plugin, as I was when I tested it. Not a blocker to merge.

jeherve replied:
I created #10696 to address this.

dereksmart reviewed (approved):
LGTM - left one non-blocking suggestion if we feel it's a good idea in a comment above.

jeherve commented:
Cherry-picked to `branch-6.8` in 3c198c35ec70832c3776d4bf803aef6444c87ad2
